[Bug] fix NPE in SQLCodeCheckInterceptor

### Search before asking

- [X] I searched the [issues](https://github.com/apache/incubator-linkis/issues) and found no similar issues.


### Linkis Component

linkis-computation-governance

### Steps to reproduce

execute the following command 
```
sh bin/linkis-cli -submitUser hadoop -engineType shell -codeType shell -code "whoami"
```

the error logs will shown
```
[INFO] User does not provide usr-configuration file. Will use default config
[INFO] user does not specify proxy-user, will use current submit-user "hadoop" by default.
[INFO] connecting to linkis gateway:http://127.0.0.1:9001
[ERROR] EXE0011,Error occured during execution: Failed to submit job
Caused by: org.apache.linkis.httpclient.exception.HttpClientResultException: errCode: 10905 ,desc: URL /api/rest_j/v1/entrance/submit request failed! ResponseBody is {"method":null,"status":1,"message":"operation failed(操作失败)s！the reason(原因)：NullPointerException: ","data":{"errorMsg":{"serviceKind":"linkis-cg-entrance","level":2,"port":9104,"errCode":0,"ip":"tv3-hadoop-06","desc":"operation failed(操作失败)s！the reason(原因)：NullPointerException: "}}}. errCode: 10905 ,desc: URL /api/rest_j/v1/entrance/submit request failed! ResponseBody is {"method":null,"status":1,"message":"operation failed(操作失败)s！the reason(原因)：NullPointerException: ","data":{"errorMsg":{"serviceKind":"linkis-cg-entrance","level":2,"port":9104,"errCode":0,"ip":"tv3-hadoop-06","desc":"operation failed(操作失败)s！the reason(原因)：NullPointerException: "}}}. ,ip: null ,port: 0 ,serviceKind: null ,ip: null ,port: 0 ,serviceKind: null
############Execute Error!!!########
```

### Expected behavior

we need consider whether ```enginer type``` is null

### Your environment

<!-- Please describe the linkis version you are using and basic environment information -->
<!-- 请描述您使用的linkis版本和基本环境信息 -->
- Linkis version used: 1.1.2
- Environment name and version:
    - cdh-5.14.2
    - hdp-3.1.5
    - hive-2.1.1
    - spark-3.2.1
    - scala-2.12.2
    - jdk 1.8.0_121
    - ....


### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

--- a/linkis-computation-governance/linkis-entrance/src/main/scala/org/apache/linkis/entrance/interceptor/impl/SQLCodeCheckInterceptor.scala
+++ b/linkis-computation-governance/linkis-entrance/src/main/scala/org/apache/linkis/entrance/interceptor/impl/SQLCodeCheckInterceptor.scala
@@ -26,8 +26,15 @@ import org.apache.linkis.manager.label.utils.LabelUtil
 class SQLCodeCheckInterceptor extends EntranceInterceptor {
 
   override def apply(jobRequest: JobRequest, logAppender: java.lang.StringBuilder): JobRequest = {
-    val engineType = LabelUtil.getCodeType(jobRequest.getLabels)
-    val languageType = CodeAndRunTypeUtils.getLanguageTypeByCodeType(engineType.toLowerCase())
+    val engineType = {
+      val engineType = LabelUtil.getCodeType(jobRequest.getLabels)
+      if (null != engineType) {
+        engineType.toLowerCase()
+      } else {
+        ""
+      }
+    }
+    val languageType = CodeAndRunTypeUtils.getLanguageTypeByCodeType(engineType)
     languageType match {
       case CodeAndRunTypeUtils.LANGUAGE_TYPE_SQL =>
         val sb: StringBuilder = new StringBuilder

--- a/linkis-computation-governance/linkis-entrance/src/main/scala/org/apache/linkis/entrance/interceptor/impl/SQLCodeCheckInterceptor.scala
+++ b/linkis-computation-governance/linkis-entrance/src/main/scala/org/apache/linkis/entrance/interceptor/impl/SQLCodeCheckInterceptor.scala
@@ -26,15 +26,15 @@ import org.apache.linkis.manager.label.utils.LabelUtil
 class SQLCodeCheckInterceptor extends EntranceInterceptor {
 
   override def apply(jobRequest: JobRequest, logAppender: java.lang.StringBuilder): JobRequest = {
-    val engineType = {
-      val engineType = LabelUtil.getCodeType(jobRequest.getLabels)
-      if (null != engineType) {
-        engineType.toLowerCase()
+    val codeType = {
+      val codeType = LabelUtil.getCodeType(jobRequest.getLabels)
+      if (null != codeType) {
+        codeType.toLowerCase()
       } else {
         ""
       }
     }
-    val languageType = CodeAndRunTypeUtils.getLanguageTypeByCodeType(engineType)
+    val languageType = CodeAndRunTypeUtils.getLanguageTypeByCodeType(codeType)
     languageType match {
       case CodeAndRunTypeUtils.LANGUAGE_TYPE_SQL =>
         val sb: StringBuilder = new StringBuilder

